sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -250,26 +238,24 @@ public HttpResponse logResponse(ClientLogger logger, HttpResponse response, Dura
      */
     private static String getRedactedUrl(URL url, Set<String> allowedQueryParameterNames) {
         String query = url.getQuery();
-
         if (CoreUtils.isNullOrEmpty(query)) {
             // URL doesn't have a query string, just return the URL as-is.
             return url.toString();
         }
 
         // URL does have a query string that may need redactions.
         // Use UrlBuilder to break apart the URL, clear the query string, and add the redacted query string.
-        // UrlBuilder urlBuilder = ImplUtils.parseUrl(url, false);
-        //
-        // CoreUtils.parseQueryParameters(query).forEachRemaining(queryParam -> {
-        //     if (allowedQueryParameterNames.contains(queryParam.getKey().toLowerCase(Locale.ROOT))) {
-        //         urlBuilder.addQueryParameter(queryParam.getKey(), queryParam.getValue());
-        //     } else {
-        //         urlBuilder.addQueryParameter(queryParam.getKey(), REDACTED_PLACEHOLDER);
-        //     }
-        // });
-
-        // return urlBuilder.toString();
-        return null;
+        UrlBuilder urlBuilder = ImplUtils.parseUrl(url, false);
```

Comment:
Heh, I think this is silly behavior from azure-core that we should clean up.

There's no need to parse the `URL` here into a `UrlBuilder`, we know which parts we capture in `UrlBuilder` which are the `scheme`, `host`, `port`, `path`, and `query`. Where all those map to a specific value getter within `URL`, and instead we can just keep track of those and the query parameters to log string separately.

This gets us out of creating and updating a backing `Map`.